tests/php/Dev/DeprecationTest.php

```diff
@@ -24,12 +24,18 @@ protected function setup(): void
         //   https://github.com/laminas/laminas-di/pull/30#issuecomment-927585210
         parent::setup();
         $this->oldHandler = set_error_handler(function (int $errno, string $errstr, string $errfile, int $errline) {
-            if (str_contains($errstr, 'SilverStripe\\Dev\\Tests\\DeprecationTest')) {
-                throw new Deprecated($errstr, $errno, '', 1);
+            if ($errno === E_USER_DEPRECATED) {
+                if (str_contains($errstr, 'SilverStripe\\Dev\\Tests\\DeprecationTest')) {
+                    throw new Deprecated($errstr, $errno, '', 1);
+                } else {
+                    // Surpress any E_USER_DEPRECATED unrelated to this unit-test
+                    return true;
+                }
```

Comment:
I won't hold back the PR for this but for future reference it's cleaner to omit the `else` in situations where the original confition block always has a `return` or `throw` statement
```suggestion
                }
                // Surpress any E_USER_DEPRECATED unrelated to this unit-test
                return true;
```